include/mbedtls/check_config.h

```diff
@@ -591,6 +591,11 @@
 #error "MBEDTLS_SSL_DTLS_ANTI_REPLAY  defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_SSL_CID) &&                              \
+    ( !defined(MBEDTLS_SSL_TLS_C) || !defined(MBEDTLS_SSL_PROTO_DTLS) )
+#error "MBEDTLS_SSL_CID  defined, but not all prerequisites"
+#endif
+
```

Comment:
Note on commit structure: in the future, we should get into the habit that every time we add something new to `config.h` in the same commit we:

- update generated files (`version_feature`, `query_config`)
- update `check_config.h` if necessary

Otherwise the commit isn't quite atomic but rather subatomic.

(That's just a note for the future though, note a request to rewrite history here.)

Reply:
Yes, though that assumes I don't forget about it, which is actually what happened most times so far. The splitting is never intentional.

Reply:
Yeah, I know, same for me. Still think it would be good for all of us to try to create that mental association "extend config.h" -> "re-generate files and consider checks".

Reply:
(Also, if we think about it before we pushed (or at least before review started), we can squash the commits together for a cleaner history.)

Reply:
Would be nice to collect these way of working comments somewhere so they don't get lost.